apps/opencs/model/world/actoradapter.cpp

```diff
@@ -113,17 +113,23 @@ namespace CSMWorld
     const std::string ActorAdapter::ActorData::getPart(ESM::PartReferenceType index) const
     {
         auto it = mParts.find(index);
-        if (it == mParts.end() && mRaceData && mRaceData->handlesPart(index))
+        if (it == mParts.end())
         {
-            if (mFemale)
+            if (mRaceData && mRaceData->handlesPart(index))
             {
-                // Note: we should use male parts for females as fallback
-                const std::string femalePart = mRaceData->getFemalePart(index);
-                if (!femalePart.empty())
-                    return femalePart;
+                if (mFemale)
+                {
+                    // Note: we should use male parts for females as fallback
+                    const std::string femalePart = mRaceData->getFemalePart(index);
+                    if (!femalePart.empty())
+                        return femalePart;
+                }
+
+                return mRaceData->getMalePart(index);
             }
 
-            return mRaceData->getMalePart(index);
+            const std::string& partName = "";
```

Comment:
For some reason Aesylwinn decided to return referende to existing string, so it will not work here.
If you want, I can change a signature of this method.

Reply:
I do not follow. The function returns a const std::string. There is no reference.

Reply:
Oh. It seems I looked to another function. Fixed now.

Reply:
For future reference (pun intended): If there would have been a reference, you would have returned a dangling reference to a temporary string. Binding a temporary to a reference does not extend the lifetime of the temporary across a return.